src/Compilers/Server/VBCSCompiler/VBCSCompiler.cs

```diff
@@ -1,24 +1,32 @@
 ﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
 
+using Microsoft.CodeAnalysis.CommandLine;
 using System;
-using System.Collections.Generic;
+using System.Collections.Specialized;
 using System.Configuration;
-using System.Diagnostics;
-using System.IO;
-using System.IO.Pipes;
-using System.Linq;
-using System.Security.AccessControl;
-using System.Security.Principal;
-using System.Threading;
-using System.Threading.Tasks;
-using System.Globalization;
-using Microsoft.CodeAnalysis.CommandLine;
-using System.Runtime.InteropServices;
 
 namespace Microsoft.CodeAnalysis.CompilerServer
 {
     internal static class VBCSCompiler 
     {
-        public static int Main(string[] args) => new DesktopBuildServerController().Run(args);
+        public static int Main(string[] args)
+        {
+            NameValueCollection appSettings;
+            try
+            {
+                appSettings = ConfigurationManager.AppSettings;
+            }
+            catch (Exception ex)
+            {
+                // It is possible for AppSettings to throw when the application or machine configuration 
+                // is corrupted.  This should not prevent the server from starting, but instead just revert
+                // to the default configuration.
+                appSettings = new NameValueCollection();
+                CompilerServerLogger.LogException(ex, "Error loading application settings");
```

Comment:
This log is for diagnostic purposes of the compiler team.  Essentially when users get into a situation where the server / msbuild and command line aren't behaving we send them the instructions on how to generate this log file.  They email it to us and we diagnose what's going on.